engine/src/main/java/org/pentaho/di/pan/PanCommandExecutor.java

```diff
@@ -0,0 +1,581 @@
+        return PanReturnCode.COULD_NOT_LOAD_TRANS.getCode();
+      } else {
+          return PanReturnCode.SUCCESS.getCode();
+      }
+    }
+
+    try {
+
+      trans.setLogLevel( getLog().getLogLevel() );
+      configureParameters( trans, params, trans.getTransMeta() );
+
+      trans.setSafeModeEnabled( YES.equalsIgnoreCase( safemode ) ); // run in safe mode if requested
+      trans.setGatheringMetrics( YES.equalsIgnoreCase( metrics ) ); // enable kettle metric gathering if requested
+
+      // List the parameters defined in this transformation, and then simply exit
+      if ( YES.equalsIgnoreCase( listParams ) ) {
+
+        printTransformationParameters( trans );
+
+        // stop right here...
+        return PanReturnCode.COULD_NOT_LOAD_TRANS.getCode(); // same as the other list options
+      }
+
+      // allocate & run the required sub-threads
+      try {
+        trans.execute( arguments );
+
+      } catch ( KettleException ke ) {
+        logDebug( ke.getLocalizedMessage() );
+        System.out.println( BaseMessages.getString( getPkgClazz(), "Pan.Error.UnablePrepareInitTrans" ) );
+        return PanReturnCode.UNABLE_TO_PREP_INIT_TRANS.getCode();
+      }
+
+      waitUntilFinished( trans, 100 ); // Give the transformation up to 10 seconds to finish execution
+
+      if ( trans.isRunning() ) {
+        getLog().logError( BaseMessages.getString( getPkgClazz(), "Pan.Log.NotStopping" ) );
+      }
+
+      getLog().logMinimal( BaseMessages.getString( getPkgClazz(), "Pan.Log.Finished" ) );
+      Date stop = Calendar.getInstance().getTime(); // capture execution stop time
+
+      int completionTimeSeconds = calculateAndPrintElapsedTime( start, stop );
+
+      if ( trans.getResult().getNrErrors() == 0 ) {
+
+        trans.printStats( completionTimeSeconds );
+        return PanReturnCode.SUCCESS.getCode();
+
+      } else {
+
+        String transJVMExitCode = trans.getVariable( Const.KETTLE_TRANS_PAN_JVM_EXIT_CODE );
+
+        // If the trans has a return code to return to the OS, then we exit with that
+        if ( !Utils.isEmpty( transJVMExitCode ) ) {
+
+          try {
+            return Integer.parseInt( transJVMExitCode );
+
+          } catch ( NumberFormatException nfe ) {
+            getLog().logError( BaseMessages.getString( getPkgClazz(), "Pan.Error.TransJVMExitCodeInvalid",
+                    Const.KETTLE_TRANS_PAN_JVM_EXIT_CODE, transJVMExitCode ) );
+            getLog().logError( BaseMessages.getString( getPkgClazz(), "Pan.Log.JVMExitCode", "1" ) );
+            return PanReturnCode.ERRORS_DURING_PROCESSING.getCode();
+          }
+
+        } else {
+            // the trans does not have a return code.
+            return PanReturnCode.ERRORS_DURING_PROCESSING.getCode();
+        }
+      }
+
+    } catch ( KettleException ke ) {
+
+      System.out.println( BaseMessages.getString( getPkgClazz(), "Pan.Log.ErrorOccurred", "" + ke.getMessage() ) );
```

Comment:
Is there a reason we are using System.out and logger

Reply:
Yes. Maintaining the way the logger took place, when it was located in `Pan.java` ( see the way it used to be, on the 8.0.0.0. branch, at https://github.com/pentaho/pentaho-kettle/blob/8.0.0.0-R/engine/src/main/java/org/pentaho/di/pan/Pan.java#L606-L607 )

Apparently, Pan's execution uses `System.out`, unless you pass along a "logfile" parameter, stating a path to a log file it would write into ( see "logfile" param at https://help.pentaho.com/Documentation/8.0/Products/Data_Integration/Command_Line_Tools )